api/rest/client/methods.go

```diff
@@ -86,20 +86,30 @@ func (c *defaultClient) Unpin(ci cid.Cid) error {
 	return c.do("DELETE", fmt.Sprintf("/pins/%s", ci.String()), nil, nil, nil)
 }
 
+// CleanPath gives a path string that can be readily appended to a url string
+func CleanPath(p string) string {
```

Comment:
ParsePath will not remove trailing slashes, but let's not do one more iteration. We won't have to care about removing slashes once we resolve https://github.com/ipfs/ipfs-cluster/issues/657. 
I am going to remove tests with paths with trailing slashes.

Reply:
Also ParsePath does much more than just cleaning path. My feeling is that client should not do anything except making the api request. 
CleanPath was a convenience or a hack until we fix the trailing slash problem, but here an invalid path would through error here. Wouldn't you expect it to call the API and let it through error with 400 or 404?

Reply:
>  Fine to leave `//` uncleaned at this point, as they're ignored by `ParsePath`.

Doesn't seem like it
```
methods_test.go:238: unexpected error /ipfs//QmbUNM297ZwxB8CfFAznK7H9YMesDoY6Tt5bPgt5MSCB2u/im.gif: path must contain at least one component
```
but anyway we don't need to clear double slashes, as it would be an invalid output provided by the user.

Reply:
Yes, it's fine. Cannot have `//` at that position, but I think later in the path it's fine.

> Wouldn't you expect it to call the API and let it through error with 400 or 404?

If the path is clearly wrong I would expect not having to trigger a request. Another option is to let this method receive an object of `path` type (forcing the user to do the parsing beforehand), but since `path` is more a utility than a canonical type that seems inconvinient.